### PR TITLE
[pickers] Replace `adapter.date` with the current `adapter.dateWithTimezone`

### PR DESCRIPTION
**Pull request by flaviendelangle**
#### Breaking changes

-  The `adapter.dateWithTimezone` method has been removed and its content has been moved the `date` method.
You can use the `adapter.date` method instead:

```diff
- adater.dateWithTimezone(undefined, 'system');
+ adater.date(undefined, 'system');
```

- The `adapter.date` method now have the behavior of the v6 `adapter.dateWithTimezone` method.
It no longer accept `any` as a value but only `string | null | undefined`

```diff
- adapter.date(new Date());
+ adapter.date();

- adapter.date(new Date('2022-04-17');
+ adapter.date('2022-04-17');

- adapter.date(new Date(2022, 3, 17, 4, 5, 34));
+ adapter.date('2022-04-17T04:05:34');

- adapter.date(new Date('Invalid Date'));
+ adapter.getInvalidDate();
```

### CODE REVIEW

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts [hunk 1]
```diff
@@ -285,17 +285,9 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs, string> {
...
-  public date = (value?: any) => {
-    if (value === null) {
-      return null;
-    }
-
-    return this.dayjs(value);
-  };
-
-  public dateWithTimezone = <T extends string | null | undefined>(
+  public date = <T extends string | null | undefined>(
```
Comment (alexfauquette) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts:288]:
```suggestion
  public date = <T extends string | null>(
```

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts [hunk 2]
```diff
@@ -285,17 +285,9 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs, string> {
...
     value: T,
```
Comment (alexfauquette) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts:289]:
```suggestion
    value?: T,
```

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts [hunk 3]
```diff
@@ -285,17 +285,9 @@ export class AdapterDayjs implements MuiPickersAdapter<Dayjs, string> {
...
-    timezone: PickersTimezone,
+    timezone: PickersTimezone = 'default',
   ): DateBuilderReturnType<T, Dayjs> => {
     type R = DateBuilderReturnType<T, Dayjs>;
```
Comment (alexfauquette) [inline packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts:292]:
```suggestion
    type R = DateBuilderReturnType<T | undefined, Dayjs>;
```

**Review by alexfauquette — COMMENTED** [review]
The fact that the first argument is mandatory is a bit strange. Is there a specific reason for that choice?

File: packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts (hunk 3 above)
Comment (flaviendelangle) [reply packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts:292]: No need for this change :+1:

**Comment by flaviendelangle** [comment]
> The fact that the first argument is mandatory is a bit strange. Is there a specific reason for that choice?

I was struggling with the typing, but I don't know why, it works fine...
Updated

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 4]
```diff
@@ -410,6 +420,34 @@ The `getYearRange` method used to accept two params and now accepts a tuple to b
...
+The `date` method now have the behavior of the v6 `dateWithTimezone` method.
+The following changes have been applied:
+
+- Its first argument is now mandatory, you need to pass `undefined` to get a date with the current timestamp instead of passing no argument
+
+  ```diff
+  - adapter.date();
+  + adapter.date();
+  ```
+
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:434]:
```suggestion
```

Not sure it's usefull to replace `adapter.date()` by `adapter.date()` XD

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 5]
```diff
@@ -410,6 +420,34 @@ The `getYearRange` method used to accept two params and now accepts a tuple to b
...
+  ```diff
+  - adapter.date(new Date());
+  + adapter.date();
+
+  - adapter.date(new Date('2022-04-17');
+  + adapter.date('2022-04-17');
+
+  - adapter.date(new Date(2022, 3, 17));
+  + adapter.date('2022-04-17');
```
Comment (alexfauquette) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:445]: Wondering if one of those two examples should have time section, such that it's clear that `.date()` supports hours, minues, ...

**Review by alexfauquette — APPROVED** [review]
Except for What I assume is a useless point in the changleog, it looks good :+1:

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md (hunk 4 above)
Comment (flaviendelangle) [reply docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:434]: :laughing: Search and replace works well but do not remove useless doc sections

**Pull request merged**